format/format.go

```diff
@@ -339,11 +348,24 @@ func trimTrailingWhitespaceInComments(comments []*ast.Comment) []*ast.Comment {
 	return comments
 }
 
-func (w *writer) writePackage(pkg *ast.Package, comments []*ast.Comment) []*ast.Comment {
+func (w *writer) writePackage(pkg *ast.Package, comments []*ast.Comment, annotations []*ast.Annotations) []*ast.Comment {
+	parsedAnnotations, _ := ast.ParseAnnotations(comments)
+
 	comments = w.insertComments(comments, pkg.Location)
 
 	w.startLine()
+
+	if len(parsedAnnotations) == 0 {
```

Comment:
When parsing annotations, corresponding comments are dropped?
There is no risk of `(len(annotations) > 0 && len(parsedAnnotations) > 0) == true`?

Reply:
> When parsing annotations, corresponding comments are dropped?

While parsing the annotations the corresponding comments are NOT dropped. In cases where the policies are optimized, we now attach the annotations to the optimized rule and module AST. But in this case there are NO comments on the optimized module (ie. the comments from the original module are not carried over to the optimized module but NOW annotations are). So in my testing I haven't come across this condition `(len(annotations) > 0 && len(parsedAnnotations) > 0) == true`.

Can you imagine a scenario where this is likely?